src/eredis_client.erl

```diff
@@ -606,9 +610,17 @@ maybe_reconnect(Reason,
         {ok, State2} ->
             {noreply, State2};
         {error, NewReason} ->
-            {noreply, schedule_reconnect(NewReason, State1)}
+            ?LOG_NOTICE("eredis: Reconnect failed: ~p", [NewReason]),
+            {noreply, schedule_reconnect(State1)}
     end.
 
+log_reconnect(retry, _Host, _Port) ->
+    %% Don't flood the log for every retry, in case the Redis node is down.
+    ok;
+log_reconnect(Reason, Host, Port) ->
+    ?LOG_WARNING("eredis: Re-establishing connection to ~p:~p due to ~p",
```

Comment:
Could we also log the `self()` pid? When this is managed via eredis_cluster, that is useful info.

Reply:
Sure. Can I drop the `eredis:` prefix too? The module name is somewhere in the log entry anyway.

Reply:
`("~p reconnecting to ~p:~p due to ~p", [self(), Host, Port, Reason])` OK?

Reply:
I tried it and I don't see module name, am I missing something?
```
1> f(),Host="127.0.0.1",Port=6379,Reason=econnrefused,logger:warning("~p reconnecting to ~p:~p due to ~p", [self(), Host, Port, Reason])
1> .
=WARNING REPORT==== 22-Jul-2022::14:29:47.860134 ===
<0.80.0> reconnecting to "127.0.0.1":6379 due to econnrefused
```

Reply:
The macros `?LOG_WARNING` etc. add file, line, mfa, etc. to the log entry metadata. See https://www.erlang.org/doc/man/logger.html#type-metadata.

To show it in the formatted log message, try something like this:

```Erlang
logger:update_handler_config(default, #{formatter => {logger_formatter, #{template => [time," ",level," at ", file, ":", line, "\n", msg,"\n"]}}}).
```

Reply:
Even the pid is in the log entry metadata.

Reply:
Thanks for explaining. I guess there is no need to explicitly add the pid then?